contribs/gmf/src/services/querymanager.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * @fileoverview This file defines the QueryManager service, which uses the
+ * c2cgeoportal's themes to configure ngeo's query service with each layer
+ * found.
+ */
+goog.provide('gmf.QueryManager');
+
+goog.require('gmf');
+goog.require('gmf.Themes');
+goog.require('ngeo.Query');
+
+
+
+/**
+ * @constructor
+ * @param {ngeo.Query} ngeoQuery The ngeo Query service.
+ * @param {gmf.Themes} gmfThemes The gmf Themes service.
+ * @param {string} gmfWmsUrl URL to the wms service to use by default.
+ * @ngInject
+ * @ngdoc service
+ * @ngname gmfThemes
+ */
+gmf.QueryManager = function(ngeoQuery, gmfThemes, gmfWmsUrl) {
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.gmfWmsUrl_ = gmfWmsUrl;
+
+  // the themes have to be loaded at this point
+  // FIXME, should it be the responsability of this manager to do this ?
+  gmfThemes.loadThemes();
```

Comment:
Okay.  The issue here is that I would have liked to use the promise to get the themes right after that, but I'll use the event instead, which will do the trick.